lib/babe/babe.go

```diff
@@ -497,6 +497,21 @@ func (b *Service) handleSlot(epoch, slotNum uint64,
 		return err
 	}
 
+	slotNumOfTheBestBlock, err := b.blockState.GetSlotForBlock(parentHeader.Hash())
+	if err != nil {
+		return fmt.Errorf("could not get slot for block %s: %w", parentHeader.Hash(), err)
+	}
+
+	if slotNumOfTheBestBlock == slotNum {
+		// pick parent of best block instead to handle slot
+		newParentHeader, err := b.blockState.GetHeader(parentHeader.ParentHash)
```

Comment:
nit rename to `grandParentHeader`? 🤔

Reply:
can't use the word grandparent. Basically we are looking for a block on which we can put our new block. So it is always parent.